pkg/operator/types/types.go

```diff
@@ -103,6 +128,62 @@ func (r Resource) GetNamespace() string {
 	return ""
 }
 
+func (r Resource) ShouldWaitForReady() bool {
+	if r.Unstructured != nil {
+		annotations := r.Unstructured.GetAnnotations()
+		if annotations == nil {
+			return false
+		}
+		waitForReady, ok := annotations[WaitForReadyAnnotation]
+		if !ok {
+			return false
+		}
+		return waitForReady == "true"
+	}
+	return false
+}
+
+func (r Resource) ShouldWaitForProperties() bool {
+	if r.Unstructured != nil {
+		annotations := r.Unstructured.GetAnnotations()
+		if annotations == nil {
+			return false
+		}
+		_, ok := annotations[WaitForPropertiesAnnotation]
+		return ok
```

Comment:
make sense.  this would return true in the case of an empty annotation value.  it would then try to parse out the key-value pairs, would find none, and would not enter the loop to wait for them.  probably a small performance gain if we return false for an empty label, so i'll make this edit